### PR TITLE
Patch tinyfilemanager to add response headers for metrics

### PR DESCRIPTION
**Pull request by benoit74**
## Rationale

https://github.com/offspot/metrics/ needs to compute the number of files added and removed in the file manager.

Current tinyfilemanager code does not permit to process URLs to detect this information, so we are now patching tinyfilemanager to add response headers when files have been added and removed, with the number of files added / removed.

## Changes
- two new global variables `$files_added` and `$files_deleted` to store amount of files added or deleted in a given request
- new `fm_report_files` function which really output the headers when needed
- new `fm_count_files` function to count files in a given folder
- a `X-TFM-Files-Added` header is returned with the number of files added (only when files have been added):
  - Upload from URL (~ lines 652-657)
  - Create a new file (~ lines 700-707)
  - Upload file  (~ lines 1050-1063), single upload (~ lines 1021-1030) and chunked upload (~ lines 1030-1041)
  - Create a Zip / Tar (~ lines 1141-1155)
  - Extract a Zip / Tar (~ lines 1210-1226)  (we needed a new `fm_count_files` function for that)
  - Mass copy + copy single file (~ lines 2395-2440)
- a `X-TFM-Files-Deleted` header is returned with the number of files deleted (only when files have been deleted):
  - Mass delete + delete single file (~ lines 2298-2316)

### CODE REVIEW

File: file-manager/Dockerfile [hunk 1]
```diff
@@ -44,6 +44,7 @@ RUN apk add --no-cache \
     && curl -L -O "$TFMPREFIX/tinyfilemanager.php" \
     && mkdir -p /data \
     && apk del curl \
+    && patch tinyfilemanager.php tinyfilemanager.php.patch \
```
Comment (rgaudin) [inline file-manager/Dockerfile:47]: patch is not installed…

Comment (rgaudin) [reply file-manager/Dockerfile:47]: and it's not copied in the image…

**Review by rgaudin — CHANGES_REQUESTED** [review]
Please test and re-request

**Comment by benoit74** [comment]
I was convinced we had test CI in place to confirm image build is OK in PRs too ; I expected to test the build locally (and fix it, because I knew it would be KO) but once I saw the branch is green after I finished to prepare the first comment with all details, I thought "oh, great, I did good job" 🤦🏻

Sorry ...

File: file-manager/tinyfilemanager.php.patch [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
++            $files_added++;
+         } else {
+             fm_set_msg(lng('Archive not created'), 'error');
+         }
+@@ -1210,6 +1226,8 @@
+ 
+         if ($res) {
+             fm_set_msg(lng('Archive unpacked'));
++            global $files_added;
++            $files_added += fm_count_files($path);
```
Comment (rgaudin) [inline file-manager/tinyfilemanager.php.patch:80]: I didn't even realize there was such a feature. I am not sure if we want to count this as Added files. You may want to discuss this with @Popolechien. Expanding can be unrelated to an _upload_ (and thus an _addition_ in user's mind). Given the tool offers easy upload of multiple files at once, uploading an Archive would make more sense as a whole.

**Review by rgaudin — APPROVED** [review]
Works like a charm ; please check the expansion thing with @Popolechien and either remove that part or merge.

File: file-manager/tinyfilemanager.php.patch (hunk 2 above)
Comment (benoit74) [reply file-manager/tinyfilemanager.php.patch:80]: We have confirmed together this is needed. E.g. if a user unzip a file with 120 files withing, this has to be counted as 120 additions.

Comment (rgaudin) [reply file-manager/tinyfilemanager.php.patch:80]: OK then 🤷‍♂️

**Pull request merged**